tests/evm/test_calldataload.py

```diff
@@ -0,0 +1,227 @@
+import pytest
+
+from typing import Optional
+from zkevm_specs.evm import (
+    Bytecode,
+    CallContextFieldTag,
+    ExecutionState,
+    RW,
+    RWTableTag,
+    StepState,
+    Tables,
+    Transaction,
+    verify_steps,
+)
+from zkevm_specs.util import rand_fp, RLC, U64
+
+TESTING_DATA = (
+    (
+        bytes.fromhex("FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF"),
+        0x20,
+        0x00,
+        bytes.fromhex("FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF"),
+        True,
+        None,
+    ),
+    (
+        bytes.fromhex("FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF"),
+        0x20,
+        0x1F,
+        bytes.fromhex("FF00000000000000000000000000000000000000000000000000000000000000"),
+        True,
+        None,
+    ),
+    (
+        bytes.fromhex("a1bacf5488bfafc33bad736db41f06866eaeb35e1c1dd81dfc268357ec98563f"),
+        0x20,
+        0x10,
+        bytes.fromhex("6eaeb35e1c1dd81dfc268357ec98563f00000000000000000000000000000000"),
+        True,
+        None,
+    ),
+    (
+        bytes.fromhex("a1bacf5488bfafc33bad736db41f06866eaeb35e1c1dd81dfc268357ec98563f"),
+        0x20,
+        0x10,
+        bytes.fromhex("6eaeb35e1c1dd81dfc268357ec98563f00000000000000000000000000000000"),
+        False,
+        0x00,
+    ),
+    (
+        bytes.fromhex("a1bacf5488bfafc33bad736db41f06866eaeb35e1c1dd81dfc268357ec98563fab"),
+        0x20,
+        0x10,
+        bytes.fromhex("aeb35e1c1dd81dfc268357ec98563fab00000000000000000000000000000000"),
+        False,
+        0x01,
+    ),
+)
+
+
+@pytest.mark.parametrize(
+    "call_data, call_data_length, offset, expected_stack_top, is_root, call_data_offset",
+    TESTING_DATA,
+)
+def test_calldataload(
+    call_data: bytes,
+    call_data_length: U64,
+    offset: U64,
+    expected_stack_top: bytes,
+    is_root: bool,
+    call_data_offset: Optional[U64],
+):
+    randomness = rand_fp()
+
+    tx = Transaction(id=1, call_data=call_data)
```

Comment:
only include `call_data` when `is_root` is true.

Reply:
Yes I might have missed this after the recent changes. Thanks for pointing out!